kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -303,38 +303,48 @@ impl<'tcx> GotocCtx<'tcx> {
                 }
             }
             (Scalar::Int(_), ty::Adt(adt, subst)) => {
-                if adt.is_struct() || adt.is_union() {
-                    // in this case, we must have a one variant ADT. there are two cases
-                    let variant = &adt.variants().raw[0];
-                    // if there is no field, then it's just a ZST
+                if adt.is_struct() {
+                    // In this case, we must have a one variant ADT. There are two cases.
+                    let variant = adt.non_enum_variant();
+                    let overall_type = self.codegen_ty(ty);
+                    // If there is no field, then it's just a ZST.
                     if variant.fields.is_empty() {
```

Comment:
is this condition really reachable considering constant evaluation returned a scalar value ?

Reply:
Looks like it's not, at least according to our regression. I've removed this `if-then-else`. Thanks for pointing this out!